src/Input/Select/Select.tsx

```diff
@@ -135,7 +135,7 @@ const Select: ISelect = (props: Props) => {
       setCursor(0);
 
       if (onChange !== undefined) {
-        return onChange(e.target.value);
+        return onChange(e);
```

Comment:
this is a breaking change, please update other projects

Reply:
why here passing `e` to onChange, and handleSelect is passing `activeElement.getAttribute('data-value')` to onChange?

Reply:
hmm, I think it's better to keep passing `e.target.value` to onChange, since the `handleSelect` can be called without a `event` object passed.

Reply:
...so do you approve of this? If you go back in the history you'll see that the above was the behaviour before the conversion into a functional component. Not sure why they did it like that, but I thought it better to restore it.

Reply:
please make sure you understand this change

Reply:
https://github.com/glints-dev/glints-aries/blob/2d402a4872e579a447ec274c67ce9a9053a1ea7c/src/Input/Select/Select.tsx#L162

```
  handleChange = (
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  ) => {
    const listener = (e: React.ChangeEvent<HTMLInputElement>) => {
      const { children } = this.props;
      const childrenArray = React.Children.toArray(children) as Array<
        React.ReactElement<SelectItemProps>
      >;

      this.setState({
        selectedValue: e.target.value,
        filterValue: childrenArray.filter(data =>
          data.props.children
            .toLowerCase()
            .includes(e.target.value.toLowerCase())
        ),
        cursor: 0,
      });

      if (onChange !== undefined) {
        return onChange(e);
      }
    };
```

This is the old onChange function. It is calling the `onChange` callback with `e`. That's why I want to revert this. In my opinion this `onChange` should not be called here at all. What's the point of calling it with what the user has typed? Ow with the change event? But for some reason it was being done like that in the past, and it did not interfere with `onChange` being called when an option is called. So to be honest, I don't understand why it was like that, but I want to revert it to that behaviour since the new one does not work, apparently.

Reply:
thank u @westwood846 got it